packages/mui-base/src/Slider/Root/useSliderRoot.ts

```diff
@@ -161,15 +177,40 @@ function useSliderRoot(parameters: UseSliderParameters): UseSliderReturnValue {
     };
   }, [id, setControlId]);
 
+  useEnhancedEffect(() => {
```

Comment:
This seems to be pretty much copied and pasted across all form components. Have you considered creating a hook encapsulating this logic?